BacktraceScreen.c

```diff
@@ -0,0 +1,527 @@
+   } else {
+      colorBasename = PROCESS_BASENAME;
+      len = xAsprintf(&informations, "Process %d: %n%s",Process_getPid(process), &indexProcessComm, process->mergedCommand.str);
+   }
+
+   RichString_appendnAscii(out, CRT_colors[DEFAULT_COLOR] | A_BOLD, informations, len);
+   if (indexProcessComm != -1) {
+      RichString_setAttrn(out, CRT_colors[colorBasename] | A_BOLD, indexProcessComm + highlightOffset, highlightLen);
+   }
+
+   free(informations);
+}
+
+static void BacktracePanelRow_displayFrame(const Object* super, RichString* out) {
+   const BacktracePanelRow* row = (const BacktracePanelRow*)super;
+   assert(row);
+   assert(row->type == BACKTRACE_PANEL_ROW_DATA_FRAME);
+
+   const BacktracePanelPrintingHelper* printingHelper = row->printingHelper;
+   const int* displayOptions = row->displayOptions;
+   const BacktraceFrameData* frame = row->data.frame;
+
+   char* functionName = frame->functionName;
+   int maxFunctionNameLength = printingHelper->maxFuncNameLen;
+   if (!!(*displayOptions & DEMANGLE_NAME_FUNCTION) &&
+         printingHelper->maxDemangledFuncNameLen > 0) {
+      maxFunctionNameLength = printingHelper->maxDemangledFuncNameLen;
+      if (frame->demangleFunctionName) {
+         functionName = frame->demangleFunctionName;
+      }
+   }
+
+   char* completeFunctionName = NULL;
+   xAsprintf(&completeFunctionName, "%s+0x%zx", functionName, frame->offset);
+
+   char* objectDisplayed = frame->objectName;
+   size_t objectLength = printingHelper->maxObjNameLen;
+   if (!!(*displayOptions & SHOW_FULL_PATH_OBJECT)) {
+      objectDisplayed = frame->objectPath;
+      objectLength = printingHelper->maxObjPathLen;
+   }
+
+   size_t maxAddrLen = printingHelper->maxAddrLen - strlen("0x");
+   char* line = NULL;
+   int objectPathStart = -1;
+
+   /*
+    * The parameters for printf are of type int.
+    * A check is needed to prevent integer overflow.
+    */
+   assert(printingHelper->maxFrameNumLen <= INT_MAX);
+   assert(maxAddrLen <= INT_MAX);
+   assert(maxFunctionNameLength <= INT_MAX);
+   assert(objectLength <= INT_MAX);
+
+   int len = xAsprintf(&line, "%*d 0x%0*zx %n%-*s %-*s",
+      (int)printingHelper->maxFrameNumLen, frame->index,
+      (int)maxAddrLen, frame->address,
+      &objectPathStart,
+      (int)objectLength, objectDisplayed ? objectDisplayed : "-",
+      (int)maxFunctionNameLength, completeFunctionName
+   );
+
+   int colors = CRT_colors[DEFAULT_COLOR];
+   if (!objectDisplayed && row->data.frame->address == 0) {
+      colors = CRT_colors[DYNAMIC_GRAY];
+   }
+
+   RichString_appendnAscii(out, colors, line, len);
+
+   if (row->settings->highlightBaseName) {
+      BacktracePanelRow_highlightBasename(row, out, line, objectPathStart);
+   }
+
+   free(completeFunctionName);
+   free(line);
+}
+
+static void BacktracePanelRow_displayError(const Object* super, RichString* out) {
+   const BacktracePanelRow* row = (const BacktracePanelRow* )super;
```

Comment:
```suggestion
   const BacktracePanelRow* row = (const BacktracePanelRow*)super;
```

Reply:
Thank you, @BenBE. I'm wondering why the CI doesn't check that? Is this something that will happen in the future?

Reply:
The code-style we use is not really set in stone, but people are encouraged to look around and follow what is around the lines they change. If you can't distinguish the code you added from what's around it, you are usually doing a good enough job with the code style. ;-)

But the style is not really strictly codified enough to automatically check it; although there's an `astyle` command listed in the styleguide that performs most of the necessary formatting work (sans some quirks).